src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -22,18 +22,26 @@ internal static unsafe class AverageFilter
         /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPerScanline, int bytesPerPixel)
+        public static void Decode(BufferSpan<byte> scanline, BufferSpan<byte> previousScanline, int bytesPerScanline, int bytesPerPixel)
         {
+            ref byte scanPointer = ref scanline.DangerousGetPinnableReference();
```

Comment:
OFF:
@KodrAus really cool to see a Rust developer around! Learning Rust is on my long list of things I want to do one time! :)